### PR TITLE
Support Swift 3 (issue #1)

### PR DESCRIPTION
**Pull request by msullivan247**
Updated the project to use Swift 3 syntax. Also renamed the `baseline` property to `lastBaseline`, to follow the change in the `NSLayoutAttribute` enum.

Switched from using Nimble to XCTest assertions, as it seems Nimble has not been updated for Swift 3 yet. Quick has a Swift 3 branch, so I'm hoping Nimble will follow shortly, then maybe we can rollback to Nimble assertions.

### CODE REVIEW

File: FluentConstraints.podspec [hunk 1]
```diff
@@ -1,13 +0,0 @@
-Pod::Spec.new do |s|
```
Comment (sjmadsen) [inline FluentConstraints.podspec:1]: I don't think you want to delete this file. It's what allows you to upload a new pod of FluentConstraints itself to CocoaPods.

Remember to bump the version (probably to 2.0) if you restore this file.

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-    - SDK="iphonesimulator9.1"
+    - SDK="iphonesimulator10.0"
   matrix:
     - DESTINATION="OS=9.1,name=iPhone 5"
     - DESTINATION="OS=9.1,name=iPhone 5s"
     - DESTINATION="OS=8.4,name=iPhone 5"
     - DESTINATION="OS=8.4,name=iPhone 5s"
 script:
-  - xctool -workspace "$WORKSPACE" -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" test
+  - xctool -project "$PROJECT" -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" test
```
Comment (sjmadsen) [inline .travis.yml]: Appears that xctool has problems with Xcode 8. A possible workaround is at https://github.com/arouanet/Whimsy/commit/7b5c98b7320ef5ba1c8ec29ca0878249ffed0332.

File: FluentConstraints/FluentConstraintSet.swift [hunk 3]
```diff
@@ -17,36 +17,36 @@ public class FluentConstraintSet {
...
 
-    public var inSuperview: FluentConstraintSet {
+    open var inSuperview: FluentConstraintSet {
         precondition(self.firstView.superview != nil, "View does not have a superview")
         self.constraints.forEach { $0.secondView = self.firstView.superview! }
         return self
     }
 
-    public func onView(view: UIView) -> FluentConstraintSet {
+    open func onView(_ view: UIView) -> FluentConstraintSet {
```
Comment (sjmadsen) [inline FluentConstraints/FluentConstraintSet.swift:38]: Might rename this one to `on(view:)` to better match Swift 3's avoidance of repeating type information in function names.

File: FluentConstraints/FluentConstraintSet.swift [hunk 4]
```diff
@@ -17,36 +17,36 @@ public class FluentConstraintSet {
...
         self.constraints.forEach { $0.secondView = view }
         return self
     }
 
-    public func asView(view: UIView) -> FluentConstraintSet {
+    open func asView(_ view: UIView) -> FluentConstraintSet {
```
Comment (sjmadsen) [inline FluentConstraints/FluentConstraintSet.swift:43]: Ditto above, `as(view:)`?

File: FluentConstraints/FluentConstraintSet.swift [hunk 5]
```diff
@@ -17,36 +17,36 @@ public class FluentConstraintSet {
...
         return onView(view)
     }
 
     // MARK: internal helpers
 
-    func fluentConstraintForView(view: UIView, attribute: NSLayoutAttribute, constant: CGFloat = 0, relation: NSLayoutRelation = .Equal) -> FluentConstraint {
+    func fluentConstraintForView(_ view: UIView, attribute: NSLayoutAttribute, constant: CGFloat = 0, relation: NSLayoutRelation = .equal) -> FluentConstraint {
```
Comment (sjmadsen) [inline FluentConstraints/FluentConstraintSet.swift:49]: `fluentConstraint(view:)`?

**Review by sjmadsen — COMMENTED** [review]
Overall 👍 once Travis is working again.

File: FluentConstraints/FluentConstraintSet.swift (hunk 4 above)
Comment (msullivan247) [reply FluentConstraints/FluentConstraintSet.swift:43]: I like this idea, but `as` is a reserved word, so the compiler won't allow me to use it as the identifier for a function. Does it make sense to update to on(view:) but keep asView(_)?

Comment (sjmadsen) [reply FluentConstraints/FluentConstraintSet.swift:43]: Looks like you can use backticks to escape the definition, and then call sites can use it straight up. (At least, that's how it appears when I did something similar in a playground.)

Point taken about the reserved word, though. Apple's advice in the fix-it is to use backticks only if the name is unavoidable. I mean, it's _always_ avoidable...

**Pull request merged**